ab3d2_source/defs.i

```diff
@@ -107,23 +79,25 @@ FrameData		equ		AlienStats+(GLF_AlienT_SizeOf_l*20)
 ObjectNames		equ		FrameData+7680
 
 ObjectStats		equ		ObjectNames+600
-O_Behaviour		equ		0
-O_GFXType		equ		2
-O_ActiveTimeout	equ		4
-O_HitPoints		equ		6
-O_ExplosiveForce equ	8
-O_Impassible	equ		10
-O_DefAnimLen	equ		12
-O_ColBoxRad		equ		14
-O_ColBoxHeight	equ		16
-O_FloorCeiling	equ		18
-O_LockToWall	equ		20
-O_ActAnimLen	equ		22
-O_SoundEffect	equ		24
-
-ObjectStatLen	equ		20*2
-
-ObjectDefAnims	equ		ObjectStats+(ObjectStatLen*30)
+
+	; Game Link File: Object Defs
+	STRUCTURE GLF_ObjT,0
+		UWORD GLF_ObjT_Behaviour_w
+		UWORD GLF_ObjT_GFXType_w
+		UWORD GLF_ObjT_ActiveTimeout_w
+		UWORD GLF_ObjT_HitPoints_w
+		UWORD GLF_ObjT_ExplosiveForce_w	; unused
+		UWORD GLF_ObjT_Impassible_w		; unused
+		UWORD GLF_ObjT_DefaultAnimLen_w	; unused
+		UWORD GLF_ObjT_CollideRadius_w
+		UWORD GLF_ObjT_CollideHeight_w
+		UWORD GLF_ObjT_FloorCeiling_w
+		UWORD GLF_ObjT_LockToWall_w		; unused
+		UWORD GLF_ObjT_ActiveAnimLen_w	; unused
+		UWORD GLF_ObjT_SFX_w
+GLF_ObjT_SizeOf_l	EQU	40
```

Comment:
I'd prefer the
LABEL xxx_SIZEOF
way of defining the size of the struct. But I can see that here we'd had to add quite a few padding bytes into the struct to get there. But I do think there's ways to manually add to the "label counter"?
Also, the offset names now get very large. What does "GLF" signify? Shouldn't ObjT be enough?

Reply:
So weirdly, when trying to set the offset for the sizeof using the offset adjustment and label macro, it was consistently miscalculating it resulting in a game crash. Not sure why because the last uword should leave the offset at 26, so just adding 14 more should align the label to 40. It doesn't seem to.

Regarding the names, following on from the notion of there being a module like namespace, GLF is GameLinkFile. Within that there are multiple structures, e.g. Alien, Object, Bullet and so on.

I don't mind changing them.

Reply:
Yeah, I don't think we need to go overboard with prefixes and suffixes. We're not going to extend this thing by large, so we'll always know what Obj_SFX_w will refers to without ambiguity, One downside of long symbol names is that the prededefined formatting will go out of whack in many places.

Reply:
I've fixed the padding issue, I don't think the offset manipulation especially liked my original invocation.
All structures that had the GLF namespace prefix are now reduced to just their basic type.